pkg/apis/core/types.go

```diff
@@ -5600,6 +5600,33 @@ const (
 	ScheduleAnyway UnsatisfiableConstraintAction = "ScheduleAnyway"
 )
 
+// NodeInclusionDirectivePolicy defines the type of node inclusion policy
+// +enum
+type NodeInclusionDirectivePolicy string
+
+const (
+	// Ignore means ignore this scheduling directive when calculating pod topology spread skew.
+	Ignore NodeInclusionDirectivePolicy = "Ignore"
+	// Respect means use this scheduling directive when calculating pod topology spread skew.
+	Respect NodeInclusionDirectivePolicy = "Respect"
+)
+
+// NodeInclusionPolicy instructs how we treat node affinity and node taints policy
+// when calculating pod topology spread skew.
+type NodeInclusionPolicy struct {
+	// NodeAffinity policy indicates how we will treat Pod's nodeAffinity/nodeSelector
+	// when calculating pod topology spread skew. Options are:
+	// - Respect (default): nodes matching nodeAffinity/nodeSelector are included.
+	// - Ignore: nodeAffinity/nodeSelector are ignored. All nodes are included.
+	NodeAffinity NodeInclusionDirectivePolicy
+	// NodeTaint policy indicates how we will treat node taints
+	// when calculating pod topology spread skew. Options are:
+	// - Respect: nodes without taints, along with tainted nodes for which the incoming pod
+	// has a toleration, are included.
+	// - Ignore (default): node taints are ignored. All nodes are included.
+	NodeTaint NodeInclusionDirectivePolicy
```

Comment:
From API reviewers, we decided that these fields should be `NodeAffinityPolicy` and `NodeTaintsPolicy`.
We will also get rid of the struct. In other words, put these 2 fields directly on `TopologySpreadConstraint`. They have to be pointers and can't be defaulted, so you need to define the behavior of `nil`.

You will need to set the ALPHA nature of the fields in each comment.

Reply:
Can I have a knowledge of the benefits by removing this struct? Placing these two fields together under one struct seems more structured to me.

Reply:
the fields are independent, so grouping them doesn't seem to add a lot, and having `nodeTaintsPolicy` reads better than `policy.nodeTaints`

Reply:
Yes, they are functionally independent, but they define the same aspect of whether nodes should be considered in calculating as we named `NodeInclusionPolicy`. Once we have more policies we need to consider like `respecting node labels`, just assumed, adding more fields parallel seems bloated.

Reply:
We shouldn't ever separate node affinity from node selectors. They are the same thing, just one API is older than the other. I don't see potential for new policies. But even if there was, since every field can be left empty, it shouldn't be much of a problem.

Reply:
Got it. I'll update as advised.